plugins/modules/site_workflow_manager.py

```diff
@@ -1763,15 +1876,95 @@ def verify_diff_deleted(self, config):
         if not site_exist:
             self.status = "success"
             msg = """The requested site '{0}' has already been deleted from the Cisco Catalyst Center and this has been
-                successfully verified.""".format(self.want.get("site_name"))
+                successfully verified.""".format(self.want.get("site_name_hierarchy"))
             self.log(msg, "INFO")
             return self
 
         self.log("""Mismatch between the playbook input for site '{0}' and the Cisco Catalyst Center indicates that
-                 the deletion was not executed successfully.""".format(self.want.get("site_name")), "INFO")
+                 the deletion was not executed successfully.""".format(self.want.get("site_name_hierarchy")), "INFO")
+        return self
+
+    def update_site_messages(self):
+        """
+        Update site messages based on the status of created, updated, and deleted sites.
+        Args:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+        Returns:
+            self (object): An instance of a class representing the status of the operation, including whether it was
+                successful or failed, any error messages encountered during operation.
+        Description:
+            This method updates the messages related to site creation, updating, and deletion in the Cisco Catalyst Center.
+            It evaluates the status of created sites, updated sites, and sites that are no longer needed for update to
+            determine the appropriate message to be set. The messages are then stored in the 'msg' attribute of the object.
+        """
```

Comment:
Can't we simplify it? Check out the below code if it works to your logic then change it, otherwise ignore..

    messages = []
    self.result['changed'] = False

    if self.created_site_list:
        messages.append("Site(s) '{}' created successfully".format(", ".join(self.created_site_list)))
        self.result['changed'] = True

    if self.updated_site_list:
        messages.append("Site(s) '{}' updated successfully".format(", ".join(self.updated_site_list)))
        self.result['changed'] = True

    if self.update_not_needed_sites:
        messages.append("Site(s) '{}' need no update".format(", ".join(self.update_not_needed_sites)))

    if self.deleted_site_list:
        messages.append("Given site(s) '{}' deleted successfully".format(", ".join(self.deleted_site_list)))
        self.result['changed'] = True

    if self.site_absent_list:
        messages.append("Unable to delete site(s) '{}' as they are not found".format(", ".join(self.site_absent_list)))

    if messages:
        self.msg = " and ".join(messages) + " in Cisco Catalyst Center."
    else:
        self.msg = "No changes were made in Cisco Catalyst Center."

    self.status = "success"
    self.result['response'] = self.msg
    self.result['msg'] = self.msg

Reply:
In few conditions, checking both create and update
Example:
if self.created_site_list and self.updated_site_list:
            self.result['changed'] = True
            if self.update_not_needed_sites:
                msg = """Site(s) '{0}' created successfully as well as Site(s) '{1}' updated successully and the some site(s)
                        '{2}' needs no update in Cisco Catalyst Center"""
                self.msg = msg.format(str(self.created_site_list), str(
                    self.updated_site_list), str(self.update_not_needed_sites))
            else:
                self.msg = """Site(s) '{0}' created successfully in Cisco Catalyst Center as well as Site(s) '{1}' updated successully in
                        Cisco Catalyst Center""".format(str(self.created_site_list), str(self.updated_site_list))
                        
 Logs:
  },
    "msg": "Site(s) '['Global/japan777']' created successfully as well as Site(s) '['Global/japan77/japan_1/cherry7', 'Global/japan77/japan_1/cherry7']' updated successully and the some site(s)\n                        '['Global/japan77/japan_1', 'Global/japan77/japan_1/cherry7']' needs no update in Cisco Catalyst Center",
    "response": "Site(s) '['Global/japan777']' created successfully as well as Site(s) '['Global/japan77/japan_1/cherry7', 'Global/japan77/japan_1/cherry7']' updated successully and the some site(s)\n                        '['Global/japan77/japan_1', 'Global/japan77/japan_1/cherry7']' needs no update in Cisco Catalyst Center"
}